### ISSUE TITLE
STM32 ADC: Monitor VBAT voltage with STM32 through voltage divider on OBC

### ISSUE THREAD

**Issue by DeflateAwning** [CalgaryToSpace/CTS-SAT-1-OBC-Firmware#175]
* Create telecommand which reports the VBAT voltage

Rev 1.4 OBC:

![Image](https://github.com/user-attachments/assets/c1a4c2c9-2d35-40eb-b4fa-aefa49bb077a)

* VBAT_MONITOR goes to Pin PC2 on STM32
* VBAT is on H2-45 and H2-26 on the stack header.

**Comment by Freddydeddydoo** [comment]
Where would I put this in the telecommand defs.c? I don't think I should put this in the eps section because I don't think I need whatever external eps module to run this. I could be wrong.

**Comment by DeflateAwning** [comment]
EPS telecommands file is probably fine. You'll also want to create a new function called something like `EPS_read_vbat_with_adc`, which could go in its own new file in the eps folder.

**Comment by Freddydeddydoo** [comment]
I'm going to put the helper function of this telecommand in the eps_commands.h header file. I don't know if that's the best place to put it because I am just using Hal ADC functions and not really getting anything straight from the EPS. However, this function will only be used for the eps so that's why I'm putting it in there.

**Comment by DeflateAwning** [comment]
Seems fine to me! Viable options are either in the EPS folder (maybe in a new file), or in the STM32 folder (e.g., where the "reset reason" files are). Not too concerned about where it goes though!

**Comment by DeflateAwning** [comment]
Any luck with this one?

**Comment by Freddydeddydoo** [comment]
Yeah, sorry. I did other PRS after this and kind of lost track of it. I will restart this PR because something went wrong, and it wouldn't build, lol. I'll get it done by this week or next.

**Comment by Freddydeddydoo** [comment]
Yea I'm starting a new branch. I wrote down all my changes so it should be fine

**Comment by Freddydeddydoo** [comment]
I had it. I forgot to clean build -_-

**Comment by DeflateAwning** [comment]
Voltage divider equation:

```
vbat_monitor = vbat * (1k / 11k)
```
